### DISCUSSION TITLE
Error: value cannot be null. Parameter name: stream (line 0).

### DISCUSSION THREAD

**Discussion by v01tt** [visose/Robots#161] [Q&A] [answered]
Hi Vincent, I tried using the URRealtime on Rhino7, latest Robots, with UR3e (via URSim; e-Series).
Got the below Error:
Error: value cannot be null. Parameter name: stream (line 0).
Need help!
![165558209-2d017dcb-18dc-4fd6-9ff1-e15d97e1b111.png](https://user-images.githubusercontent.com/77627544/169638762-3104a222-9038-4c93-a3c5-2d761c056b36.png)

**Comment by visose** [comment]
Hi, can you share the code inside the scripting component?

**Reply by visose** [reply]
I just tested it and it works as expected. Although it looks like UR has deprecated the real-time interface favor of RTDE. I'll check if I can add this. I remember there were several issues with the real-time interface.

**Comment by visose** [comment, marked as answer]
The latest version can read outputs using [Real-Time Data Exchange](https://www.universal-robots.com/articles/ur/interface-communication/real-time-data-exchange-rtde-guide/), which replaces the older Realtime interface.
Here's an example .gh file on how to use it: [urrealtimedataexchange.zip](https://github.com/visose/Robots/files/8825729/urrealtimedataexchange.zip)
